content/blog/2018/04/2018-04-10-opinionated-cd-jenkins-x.adoc

```diff
@@ -0,0 +1,245 @@
+
+The right-hand side is the continuous delivery pipeline.
+This kicks in when the application is ready to be updated with a new release.
+This is the “GitOps” repo behind the scenes that controls the state of things in Kubernetes.
+A promotion on this side is a pull request, and then a merge,
+from the staging repository to the production repository.
+
+== Installing Jenkins X
+
+The jx command line has a jx install command that installs it into a Kubernetes cluster.
+
+The best experience initially is using Google’s excellent GKE service:
+
+[source]
+----
+> jx create cluster gke
+----
+
+This will ask a few questions, and go and set it all up for you in a
+cluster set aside for Jenkins X (recommended).
+Jenkins X runs entirely as services on top of a Kubernetes cluster.
+
+[source]
+----
+> jx install
+----
+
+Is designed to work with a Kubernetes cluster (if it already exists,
+recommendation is to have a cluster put aside for Jenkins X if possible).
+Amazon EKS support is coming (mostly it is around testing),
+that service is in beta/early access so it is still a work in progress,
+as is Microsoft Azures excellent AKS service.
+
+== So where is Jenkins?
+
+Good question, thanks for asking. Well, it is behind the scenes.
+As you have seen, there was no direct interaction with Jenkins,
+but it is there, running the pipelines for continuous integration and
+continuous delivery of the respective repositories, and orchestrating things with Kubernetes.
+
+If you run `jx get pipelines` you can see URLs to the various pipelines
+that have been setup for you are part of interacting with Jenkins X.
+
+By the way,
+link:https://jenkins.io/blog/2018/03/19/introducing-jenkins-x/[James Strachan has written an extensive blog on jenkins.io]
+that really explores the Jenkins X project in-depth.
+Once you finish reading this blog, take a stroll on over there and read James'.
+He also provides several ways you can get involved in the project.
+
+== What else can I do with the command line?
+
+Lots, the jx command line has built in help:
+
+`jx open`
+
+     - open apps, services or pipelines in your browser
+
+`jx activity`
+
+     - explains how things got to where they are, a history
+
+`jx get environments`
+
+     - list environments
+
+`jx get apps`
+
+     - show the state of applications, what versions are in what environments.
+
+== What’s next
+
+There is a whole lot more to this, and lots more moving parts and services
+that are set up for you that are very useful, but it is best to head over
+to link:http://jenkins-x.io/[jenkins-x.io] and have a look.
+
+This project is obviously in early stages (it is stll a Draft JEP after all) and there is lots happening.
+Check out the link:http://jenkins-x.io/community/[Jenkins X community]
+if you want to chat on slack, IRC, issues or email.
+Also, read the
+link:https://github.com/jenkinsci/jep/blob/master/jep/400/README.adoc[Jenkins Enhancement Proposal doc].
+
+CloudBees recently joined the
```

Comment:
I think can drop this paragraph if not relevant (made sense when blog was hosted on blog.cloudbees.com but probably less so here)